specifyweb/backend/stored_queries/batch_edit.py

```
@@ -1095,7 +1095,21 @@ def _get_orig_column(string_id: str):
     # Rather than sifting the data, we just add a default visual order.
-    visual_order = Func.first(headers_enumerated)
+    visual_order_groups: list[list[int]] = [[] for _ in visible_fields]
+    for index, (key, _header) in headers_enumerated:
+        # Find the column's original position if it existed in the origin query
+        original_place = key[0]
+        duplicate_index = key[1]
+        if original_place < len(visible_fields):
+            visual_order_groups[original_place].insert(duplicate_index, index)
+        else:
+            # New field/column. Add it to the end of the dataset.
+            if existing_column is None:
```

Ruff (0.15.12)

[error] Undefined name `existing_column`

(F821)